flask2postman/postman_v1.py

```diff
@@ -1,46 +1,59 @@
-#!/usr/bin/env python
 from __future__ import print_function
 
 import re
-import os
-import site
-import sys
-from importlib import import_module
 from time import time
 from uuid import uuid4
 
-__version__ = "1.4.3"
+from .utils import trim
 
 methods_order = ["GET", "POST", "PUT", "PATCH", "DELETE", "COPY", "HEAD",
                  "OPTIONS", "LINK", "UNLINK", "PURGE"]
 
 var_re = re.compile(r"(?P<var><([a-zA-Z0-9_]+:)?(?P<var_name>[a-zA-Z0-9_]+)>)")
 
-venv_warning = ("WARNING: Attempting to work in a virtualenv. If you encounter "
-                "problems, please install flask2postman inside the virtualenv.")
-
-
-PY2 = sys.version_info[0] < 3
-if PY2:
-    maxint = sys.maxint
-else:
-    maxint = sys.maxsize
-
 
 def get_time():
     return int(round(time() * 1000))
 
 
 class Collection:
 
-    def __init__(self, name):
+    def __init__(self, args):
         self._folders = []
         self._requests = []
+        self._args = args
 
         self.id = str(uuid4())
-        self.name = name
+        self.name = args.name
         self.timestamp = get_time()
 
+    def add_rules(self, current_app):
+        for rule in current_app.url_map.iter_rules():
+            if rule.endpoint == "static" and not self._args.static:
+                continue
+
+            folder = None
+            if self._args.folders:
+                try:
+                    blueprint_name, _ = rule.endpoint.split('.', 1)
+                except ValueError:
+                    pass
+                else:
+                    folder = self.get_folder(blueprint_name)
+
+            endpoint = current_app.view_functions[rule.endpoint]
+            description = trim(endpoint.__doc__)
+
+            for method in rule.methods:
+                if method in ["OPTIONS", "HEAD"] and not self._args.all:
+                    continue
+
+                request = Request.from_werkzeug(rule, method, self._args.base_url)
+                request.description = description
+                if self._args.folders and folder:
+                    folder.add_request(request)
+                self.add_request(request)
+
```

Comment:
Great idea to move that logic here!

Can we just rename it to `add_rule` (without `s`) and modify it accordingly, so that we are not coupled to a Flask application object? We can go the simple route and pass the werkzeug rule object, but that would be even better to just pass all the arguments needed directly to the method, so that we do not close the door of supporting other frameworks than Flask someday.

Reply:
Also, that would require to move the folder creation logic out of here, and pass the folder object as an argument.

Reply:
Or maybe we could just do like the `Request` class and implement a `from werkzeug` class method that instantiate the object at the same time. That way, we could couple the implementation with the `werkzeug.Map` object, without definitely enforcing Flask (or Werkzeug) as the source framework.